osu.Game/Overlays/BeatmapSet/BeatmapBadgePill.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.LocalisationExtensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Localisation;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+
+#nullable enable
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public abstract class BeatmapBadgePill : CompositeDrawable
+    {
+        [Resolved]
+        protected OsuColour Colours { get; private set; } = null!;
+
+        [Resolved(canBeNull: true)]
+        protected OverlayColourProvider? ColourProvider { get; private set; }
+
+        /// <summary>
+        /// The text displayed on the badge's label.
+        /// </summary>
+        public abstract LocalisableString BadgeText { get; }
+
+        /// <summary>
+        /// The colour of the badge's label.
+        /// </summary>
+        public abstract Colour4 BadgeColour { get; }
+
+        // todo: add linking support, to allow redirecting featured artist badge to corresponding track and spotlight badge to wiki page.
```

Comment:
This comment is likely misplaced. The explicit badge won't have linking probably ever, which means I'm like 90% sure that linking support should be added to the subclasses and not the common superclass.

Reply:
Wasn't sure whether to put this in two different places or one shared place, and I honestly think neither matter since they'll either find it here or there by search, but sure.